### PR TITLE
[WINDOWS][MSVC] Fix MSVC warnings

### PR DESCRIPTION
**Pull request by tqchen**
This PR fixes various warnings bought by MSVC.

TODO: deprecate `__tvm_main__` symbol and update
testcase so windows works as normal.

### CODE REVIEW

**Comment by tqchen** [comment]
cc @tmoreau89 @rkimball

**Comment by junrushao1994** [comment]
It looks very clean now!

I actually found the unicode in data_layout.h: they are in line 317 and 318, which is "…"

File: CMakeLists.txt [hunk 1]
```diff
@@ -109,6 +109,22 @@ if(MSVC)
       endif(${flag_var} MATCHES "/MD")
     endforeach(flag_var)
   endif()
+  # Disable common MSVC warnings
+  # Integer conversion warnings(e.g. int64 to int)
+  add_compile_options(/wd4244)
```
Comment (junrushao1994) [inline CMakeLists.txt:114]: we might revisit to cmake file to avoid global flags, but for now it is good enough

**Review by tmoreau89 — APPROVED** [review]
Thank you @tqchen I've tested it on a windows machine and it addresses some issues

**Review by junrushao1994 — APPROVED** [review]

**Comment by tmoreau89** [comment]
Thank you @junrushao1994 @ZihengJiang @tqchen , the PR has been merged

**Pull request merged**